---
title: "gene2peptide"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{gene2peptide}
  %\VignetteEngine{rmarkdown::pdf_document}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r setup}
library(gene2protein)
```

# Repository

<https://github.com/rforbiodatascience23/group21package>

# Functions

### Function 1

Takes an integer input, that specifies the desired length of the sequence, and generates a random DNA sequence of a specified length, using the nucleotide bases **adenine (A), cytosine (C), guanine (G) and thymine (T)**.

```{r}
dna <- random_peptide(300)
dna
```

### Function 2 - convert_t\_to_u()

The function `convert_t_to_u` takes a DNA sequence as input and returns the corresponding RNA sequence where thymine (T) has been replaced with uracil (U). The function achieves this by using the `gsub` function to replace all occurrences of "T" with "U" in the DNA sequence. This function can be used to convert a DNA sequence into an RNA sequence by substituting thymine with uracil. It is important to note that the function assumes the input DNA sequence is valid and does not perform any error checking.

```{r}
rna <- convert_t_to_u(dna)
rna
```

### Function 3

The purpose of the function is to split a given RNA sequence into codons, from the specified start position. The function then returns a vector of substrings (codons). NOTE: The function has no input-control meaning that any non-RNA string can be given and the function will still split it.

```{r}
split_codons <- codons(rna)
split_codons
```

### Function 4

Takes a codon and converts it to an amino acid shortcode.

```{r}
aa_seq <- codon_to_aa(split_codons)
aa_seq
```

### Function 5

Finds unique amino acids in the peptide and creates a barplot for the number of occurrences for each amino acid.

```{r}
aa_content(aa_seq)
```

# Discussion

### Use cases

Our package covers every step in the central dogma from the creation of sample DNA to the translation of peptide sequences.

### Additional functions

Rather than inputting a sample sequence, the function could take as input, a fasta file of DNA sequence data, and output a fasta file containing the translated peptide data.
