---
title: "gene2peptide"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{gene2peptide}
  %\VignetteEngine{rmarkdown::pdf_document}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r setup}
library(gene2protein)
```

# Repository

<https://github.com/rforbiodatascience23/group21package>

# Functions

### Function 1

Takes an integer input, that specifies the desired length of the sequence, and generates a random DNA sequence of a specified length, using the nucleotide bases **adenine (A), cytosine (C), guanine (G) and thymine (T)**.

```{r}
dna <- random_peptide(300)
dna
```

### Function 2

Converts DNA to RNA by substituting Thymine with Uracil.

```{r}
rna <- convert_t_to_u(dna)
rna
```

### Function 3 - codons()

The intended application for the function is to split a given RNA sequence into codons from the specified start position in line with translation of RNA. The function then returns a vector of codons

Function Operation: The input for the function is a string and a integer for overall start position. Based on the length of the string and the start position, the function will create a vector of substrings with the function substring(). The substring() function operates by creating two vectors; one for the start position and one for the stop position for each codon and splitting the string based on each set of start/stop position.
NOTE: The function has no input-control and as consequence any non-RNA string can be given and the function will still split it. 

The following example displays calling the function with the variable created by function 2 and the output of the function

```{r}
split_codons <- codons(rna)
split_codons
```

### Function 4

Takes a codon and converts it to an amino acid shortcode.

```{r}
aa_seq <- codon_to_aa(split_codons)
aa_seq
```

### Function 5

Finds unique amino acids in the peptide and creates a barplot for the number of occurrences for each amino acid.

```{r}
aa_content(aa_seq)
```

# Discussion

### Use cases

Our package covers every step in the central dogma from the creation of sample DNA to the translation of peptide sequences.

### Additional functions

Rather than inputting a sample sequence, the function could take as input, a fasta file of DNA sequence data, and output a fasta file containing the translated peptide data.
