---
title: "gene2peptide"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{gene2peptide}
  %\VignetteEngine{rmarkdown::pdf_document}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r setup}
library(gene2protein)
```

# Functions

### Function 1

Creates a random DNA sequence of a given length.

```{r}
#Testing function 1
sample_(300)
```

### Function 2

Converts DNA to RNA by substituting Thymine with Uracil.

```{r}

```

### Function 3

The purpose of the function is to split a given RNA sequence into codons, from the specified start position. The function then returns a vector of substrings (codons).
NOTE: The function has no input-control meaning that any non-RNA string can be given and the function will still split it. 

### Function 4

Takes a codon and converts it to an amino acid shortcode.

### Function 5

Finds unique amino acids in the peptide and creates a barplot for the number of occurrences for each amino acid.

# Discussion

### Use cases

Our package covers every step in the central dogma from the creation of sample DNA to the translation of peptide sequences.

### Additional functions

Rather than inputting a sample sequence, the function could take as input, a fasta file of DNA sequence data, and output a fasta file containing the translated peptide data.
